### PR TITLE
Global Imports

### PR DESCRIPTION
**Pull request by mikearnaldi**
cc @0x706b [x] missing transformers to add the imports and [x] extension of the language server to autocomplete the globally available symbols

(from @0x706b)
Adds the ability to define a project-specific, global "prelude". Marking an import as `@tsplus global` will allow the symbols imported by that declaration to be available project-wide.

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -44863,6 +44948,30 @@ namespace ts {
...
+        function tryCacheTsPlusGlobalSymbol(declaration: ImportDeclaration): void {
+            if (declaration.importClause && declaration.importClause.namedBindings && isNamedImports(declaration.importClause.namedBindings) && isStringLiteral(declaration.moduleSpecifier)) {
+                const location = declaration.moduleSpecifier.text;
+                const tags = collectTsPlusGlobalTags(declaration);
+
+                if (tags.length > 0) {
+                    if (pathIsRelative(location)) {
+                        error(declaration, Diagnostics.Global_import_path_cannot_be_relative);
+                        return;
+                    }
```
Comment (0x706b) [inline src/compiler/checker.ts:44960]: For now, relative paths are forbidden with a diagnostic.

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -3216,9 +3216,7 @@ namespace ts {
         }
 
         function checkAndReportErrorForUsingTsPlusTypeAsValue(errorLocation: Node, name: __String, meaning: SymbolFlags, globalImport: TsPlusGlobalImport): boolean {
-            if (meaning & (SymbolFlags.Value & ~SymbolFlags.NamespaceModule) &&
-                meaning !== SymbolFlags.Type &&
-                meaning !== (SymbolFlags.Type | SymbolFlags.Namespace)) {
+            if ((meaning & SymbolFlags.Value) === SymbolFlags.Value) {
```
Comment (0x706b) [inline src/compiler/checker.ts:3219]: For future reference: I'm pretty but this works, but if inexplicable/incorrect errors saying `'x' only refers to a type, but is being used as a value here` crop up from using global imports, it's probably this flag check that needs to be modified. Bit-flags are strange.

Comment (mikearnaldi) [reply src/compiler/checker.ts:3219]: indeed bitmask based checks are always hard

Comment (mikearnaldi) [reply src/compiler/checker.ts:3219]: Doing `A & Mask === Mask` is what I used to do in the past, it always felt strange to me that TS is checking usually like `if (X & Mask) {}` and don't fully understand it

**Pull request merged**